File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        l, r= 0, len(s)-1
        res = list(s)
        while l < r:
            if res[l].lower() in "aeiou":
                if res[r].lower() in "aeiou":
                    res[l], res[r] = res[r], res[l]
                    l += 1
                    r -= 1
                else:
                    r -= 1
            else:
                l += 1
        return ''.join(res)

File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.cpp
class Solution {
    bool check(char a){
        return a == 'a' || a == 'e' || a == 'i' || a == 'o' || a == 'u';
    }
public:
    string reverseVowels(string s) {
        int n = s.size();
        int l=0, r=n-1;
        while(l<r){
            while(l<r && !check(tolower(s[l]))) ++l;
            while(l<r && !check(tolower(s[r]))) --r;
            swap(s[l++], s[r--]);
        }
        return s;
    }
};